### PR TITLE
Fix for Factorio 1.0.0 and some mods

### LINKED ISSUE #47: Error when loading 1.0 with Bob's mods
Error occurs when starting up the application. No item will load following the error.

`See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at Foreman.DataCache.ReadModSettings() in E:\Projects\ForemanGit\Foreman\DataCache.cs:line 392
   at Foreman.DataCache.LoadAllData(List`1 enabledMods, CancellableProgress progress) in E:\Projects\ForemanGit\Foreman\DataCache.cs:line 223
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Foreman.ProgressForm.<ProgressForm_Load>d__4.MoveNext() in E:\Projects\ForemanGit\Foreman\Forms\ProgressForm.cs:line 44
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()

************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
Foreman
    Assembly Version: 0.0.1.1
    Win32 Version: 0.0.0.0
    CodeBase: file:///D:/Downloads/Release/Foreman.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Microsoft.GeneratedCode
    Assembly Version: 1.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
NLua
    Assembly Version: 1.3.2.0
    Win32 Version: 1.3.2
    CodeBase: file:///D:/Downloads/Release/NLua.DLL
----------------------------------------
KeraLua
    Assembly Version: 1.3.2.0
    Win32 Version: 1.3.2.0
    CodeBase: file:///D:/Downloads/Release/KeraLua.DLL
----------------------------------------
Newtonsoft.Json
    Assembly Version: 10.0.0.0
    Win32 Version: 10.0.1.20720
    CodeBase: file:///D:/Downloads/Release/Newtonsoft.Json.DLL
----------------------------------------
Microsoft.CSharp
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/Microsoft.CSharp/v4.0_4.0.0.0__b03f5f7f11d50a3a/Microsoft.CSharp.dll
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
System.Dynamic
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Dynamic/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Dynamic.dll
----------------------------------------
Anonymously Hosted DynamicMethods Assembly
    Assembly Version: 0.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/mscorlib/v4.0_4.0.0.0__b77a5c561934e089/mscorlib.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
System.Runtime.Serialization
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll
----------------------------------------
System.IO.Compression.FileSystem
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.IO.Compression.FileSystem/v4.0_4.0.0.0__b77a5c561934e089/System.IO.Compression.FileSystem.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.`

### PR DESCRIPTION
**Pull request by ybeauchamph**
This allow loading with mods on Factorio 1.0.0 

I'm still having issue #43 with missing recipe with Krastorio2.
Example recipe: iron plate recipe in-game is 5 plates for 10 ores (16s), but it shows up as 1 plate for 1 ore (3.2s) in Foreman.

It should fix #47

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,5 @@
 /.vs/
 /Foreman/Foreman.csproj.user
-/Foreman/bin/
-/Foreman/obj/
-/ForemanTest/obj/
 /packages/
-/ForemanTest/bin/Release/
+[Bb]in/
```
Comment (Rybadour) [inline .gitignore]: Please don't make such wide spread changes to the .gitignore. I'd rather see the originals kept as they target specific subdirectories. Was there a reason why this was necessary?

File: Foreman/Mod.cs [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
 
 namespace Foreman
 {
 	public class Mod
 	{
-        public String Id = "";
-		public String Name = "";
-		public String title = "";
-		public String version = "";
+		public string Id = "";
```
Comment (Rybadour) [inline Foreman/Mod.cs:9]: I'm no expert in C# so I'm probably just ignorant but what's the purpose behind changing these to "string"?

**Review by Rybadour — CHANGES_REQUESTED** [review]
Awesome change thank you! Please just fix up the .gitignore and we can promptly merge this.

File: Foreman/Mod.cs (hunk 2 above)
Comment (ybeauchamph) [reply Foreman/Mod.cs:9]: `string` is just an alias for `System.String` just like `int` is an alias to `System.Int32`.
It doesn't change anything, I'm just used to string and most code in C# use string for the type and String for the methods.
I will revert back to use String here though for consistency in the project.

File: .gitignore (hunk 1 above)
Comment (ybeauchamph) [reply .gitignore]: I changed this to ignore files from ForemanTest/bin/Debug/

**Pull request merged**